pxtblocks/blocklyloader.ts

```diff
@@ -788,51 +843,141 @@ namespace pxt.blocks {
-                this.setColour(230);
-                this.appendDummyInput()
-                    .appendField("absolute of");
-                this.appendValueInput("x")
-                    .setCheck("Number")
-                this.setInputsInline(true);
-                this.setOutput(true, "Number");
-                this.setTooltip(lf("Math operators."));
+                this.jsonInit({
+                  "message0": lf("absolute of %1"),
+                  "args0": [
+                    {
+                      "type": "input_value",
+                      "name": "x",
+                      "check": "Number"
+                    }
+                  ],
+                  "inputsInline": true,
+                  "output": "Number",
+                  "colour": blockColors['math']
+                });
+
+                setHelpResources(this,
+                    'math_op3',
+                    lf("absolute number"),
+                    lf("absolute value of a number"),
+                    '/blocks/math/abs'
+                );
             }
         };
 
+        // pxt device_random
         Blockly.Blocks['device_random'] = {
             init: function () {
-                this.setHelpUrl('/blocks/math/random');
-                this.setColour(230);
-                this.appendDummyInput()
-                    .appendField("pick random 0 to");
-                this.appendValueInput("limit")
-                    .setCheck("Number")
-                    .setAlign(Blockly.ALIGN_RIGHT);
-                this.setInputsInline(true);
-                this.setOutput(true, "Number");
-                this.setTooltip(lf("Returns a random integer between 0 and the specified bound (inclusive)."));
+                this.jsonInit({
+                  "message0": lf("pick random 0 to %1"),
+                  "args0": [
+                    {
+                      "type": "input_value",
+                      "name": "limit",
+                      "check": "Number"
+                    }
+                  ],
+                  "inputsInline": true,
+                  "output": "Number",
+                  "colour": blockColors['math']
+                });
+
+                setHelpResources(this,
+                    'device_random',
+                    lf("pick random number"),
+                    lf("Returns a random integer between 0 and the specified bound (inclusive)."),
+                    '/blocks/math/random'
+                );
             }
         };
+
+        // builtin math_number
+        //XXX Integer validation needed.
+        installHelpResources(
+            'math_number',
+            lf("number"),
+            (pxt.appTarget.compile && pxt.appTarget.compile.floatingPoint) ? lf("a decimal number") : lf("an integer number"),
+            '/blocks/math/random'
+        );
+
+        // builtin math_arithmetic
+        let msg:any = Blockly.Msg;
+        msg.MATH_ADDITION_SYMBOL = lf("_op_add");
```

Comment:
Same as ``_for_do`` and ``_while_do``, please open an issue and we'll fix this in a future commit.